File: frontend/src/types/code-review.ts
export interface ReviewIssueItem {
  id: string;
  tool: 'Semgrep' | 'SonarQube' | 'Bandit' | 'ESLint' | 'Pylint';
  type: 'VULNERABILITY' | 'CODE_SMELL' | 'DEAD_CODE' | 'NAMING' | 'COMPLEXITY';
  severity: 'CRITICAL' | 'HIGH' | 'MEDIUM' | 'LOW';
  file_path: string;
  line_number: number;
  code_snippet: string;
  ai_explanation: string;
  ai_suggestion: string;
  patch_diff?: string;
}

export interface CodeReviewResponse {
  id: string;
  repository_id: string;
  overall_score: number;
  grade: string;
  cyclomatic_complexity: number;
  maintainability_index: number;
  dead_code_count: number;
  naming_violations_count: number;
  code_smells_count: number;
  issues: ReviewIssueItem[];
  reviewed_at: string;
}

File: backend/app/schemas/code_review.py
import uuid
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel

class ReviewIssueItem(BaseModel):
    id: str
    tool: str # 'Semgrep' | 'SonarQube' | 'Bandit' | 'ESLint' | 'Pylint'
    type: str # 'VULNERABILITY' | 'CODE_SMELL' | 'DEAD_CODE' | 'NAMING' | 'COMPLEXITY'
    severity: str # 'CRITICAL' | 'HIGH' | 'MEDIUM' | 'LOW'
    file_path: str
    line_number: int
    code_snippet: str
    ai_explanation: str
    ai_suggestion: str
    patch_diff: Optional[str] = None

class CodeReviewResponse(BaseModel):
    id: uuid.UUID
    repository_id: uuid.UUID
    overall_score: int
    grade: str
    cyclomatic_complexity: float
    maintainability_index: float
    dead_code_count: int
    naming_violations_count: int
    code_smells_count: int
    issues: List[ReviewIssueItem]
    reviewed_at: datetime

    class Config:
        from_attributes = True
